pkg/storegateway/series_refs.go

```diff
@@ -1151,15 +1153,14 @@ func (s *loadingSeriesChunkRefsSetIterator) singlePassStringify(symbolizedSet sy
 	// This can be released by the caller because loadingSeriesChunkRefsSetIterator doesn't retain it after Next() is called again.
 	set := newSeriesChunkRefsSet(len(symbolizedSet.series), true)
 
-	labelsBuilder := labels.NewScratchBuilder(maxLabelsPerSeries)
```

Comment:
can you also remove the accumulation of `maxLabelsPerSeries`?

From what I remember it was it was added to reduce reallocations of the slices in ScratchBuilder. But on another look now I don't see much benefit from that: there's additional allocations on every `.Labels()` call for every series. So saving on a few resizings isn't going to tilt the scales. 

If that makes sense to you too we can remove `maxLabelsPerSeries`; otherwise there's `BenchmarkOpenBlockSeriesChunkRefsSetsIterator` which might help with the decision.